mustang_codegen/lib/src/app_aspect_generator.dart

```diff
@@ -0,0 +1,143 @@
+import 'package:analyzer/dart/element/element.dart';
+import 'package:build/build.dart';
+import 'package:mustang_codegen/src/codegen_constants.dart';
+import 'package:mustang_codegen/src/aspect_hook_generator.dart';
+import 'package:mustang_codegen/src/utils.dart';
+import 'package:mustang_core/mustang_core.dart';
+import 'package:path/path.dart' as p;
+import 'package:source_gen/source_gen.dart';
+
+class AppAspectGenerator extends Generator {
+  @override
+  String generate(LibraryReader library, BuildStep buildStep) {
+    Iterable<AnnotatedElement> aspects =
+        library.annotatedWith(const TypeChecker.fromRuntime(Aspect));
+    StringBuffer aspectBuffer = StringBuffer();
+    if (aspects.isEmpty) {
+      return '$aspectBuffer';
+    }
+
+    aspectBuffer.writeln(_generate(
+      aspects.first.element,
+      aspects.first.annotation,
+      buildStep,
+    ));
+
+    return '$aspectBuffer';
+  }
+
+  String _generate(
+    Element element,
+    ConstantReader annotation,
+    BuildStep buildStep,
+  ) {
+    _validate(element, annotation);
+
+    String aspectName = element.displayName;
+    String generatedAspectName = element.displayName.replaceFirst(r'$', '');
+    String generatedAspectNameWithLowerCase =
+        '${generatedAspectName[0].toLowerCase()}${generatedAspectName.substring(1)}';
+    String importAspect = p.basenameWithoutExtension(buildStep.inputId.path);
+
+    String pkgName = buildStep.inputId.package;
+
+    List<MethodElement> invokeHooksMethods = [];
+    List<String> imports = [];
+
+    element.visitChildren(AspectHookGenerator(
+      invokeHooksMethods,
+      imports,
```

Comment:
aspecthookgenerator is not touch this imports

Reply:
removing